rpc/src/rpc_subscriptions.rs

```diff
@@ -1411,6 +1411,12 @@ pub(crate) mod tests {
                     encoding: UiAccountEncoding::Binary,
                 }));
 
+            // Sleep here to ensure adequate time for the async thread to fully process the
+            // subscribed notification before the bank transaction is processed. Without this
+            // sleep, the bank transaction ocassionally completes first and we hang forever
+            // waiting to receive a bank notification.
+            std::thread::sleep(Duration::from_millis(100));
```

Comment:
Possible to poll here on the subscribe notification getting processed?

Reply:
Would be better than naked sleep.

Reply:
I don't believe there's existing instrumentation we can poll to check for completion (but I would love to be wrong!).

We can certainly add the instrumentation, but I'm not crazy about it. It adds a fair amount of complexity for something we would only use for unit tests

Reply:
I think the `assert_subscribed` method is about the best we have. I wonder if there's some ordering in subscription creation that should be tweaked so that we only pass that assert when everything is completed.
Brennan, could you tell where in the subscription we get to when the tx is processed, in the failure cases?

Reply:
I'm not sure exactly how far we get, but I know we didn't make it to the Subscribed notification type match because we didn't hit the print I placed there. I'm sure it varies run-to-run due to the nature of being async

Reply:
Thanks. For sure it will vary, I was just curious what you saw.
Yeah, I am thinking if we can rework the subscription registration so that the map entry checked by the assert is the last thing that happens, we could make the assert actually useful. I'll take a look at some point, and we can make this test more robust, if that approach works.
Appreciate you following up on this!

Reply:
> Thanks. For sure it will vary, I was just curious what you saw.
> 
> Yeah, I am thinking if we can rework the subscription registration so that the map entry checked by the assert is the last thing that happens, we could make the assert actually useful. I'll take a look at some point, and we can make this test more robust, if that approach works.
> 
> Appreciate you following up on this!

Sounds good! Thank you!